pkg/operator/ceph/object/controller.go

```diff
@@ -142,9 +144,67 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		}
 	}
 
+	// Watch Secrets secrets annotated for the object store
+	err = c.Watch(source.Kind[client.Object](mgr.GetCache(),
+		&corev1.Secret{TypeMeta: metav1.TypeMeta{Kind: "Secret", APIVersion: corev1.SchemeGroupVersion.String()}},
+		handler.EnqueueRequestsFromMapFunc(mapSecretToCR),
+		secretPredicate()))
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+// Watch update and create secrets annotated with the object store name
+func secretPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			secret, ok := e.ObjectNew.(*corev1.Secret)
+			if !ok {
+				return false
+			}
+			annotations := secret.GetAnnotations()
+			return annotations[objectStoreDependentResourceAnnotation] != ""
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			secret, ok := e.Object.(*corev1.Secret)
+			if !ok {
+				return false
+			}
+			annotations := secret.GetAnnotations()
+			return annotations[objectStoreDependentResourceAnnotation] != ""
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false // Ignore deletes for simplicity
+		},
```

Comment:
It might be a good thing to reconcile when the secret is deleted, in order to catch the case where a user has deleted something that they shouldn't have.

```suggestion
		DeleteFunc: func(e event.DeleteEvent) bool {
			return true // potentially catch secret deleted when still referenced
		},
```

Reply:
done